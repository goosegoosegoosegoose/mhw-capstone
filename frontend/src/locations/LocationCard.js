import React from "react";
import { Card } from "react-bootstrap";
import { useNavigate } from "react-router-dom";

const LocationCard = ({id, name, icon}) => {
  const nav = useNavigate();

  const handleClick = () => {
    nav(`/locations/${id}`)
  }

  return (
    <Card className="my-3 mx-3" onClick={handleClick} style={{width: '30vw', cursor: "pointer"}} bg="dark">
      <div className="row justify-content-center">
        <div className="col-sm-11 pt-2 mt-2">
          {icon ? <Card.Img style={{cursor: "pointer"}} onClick={handleClick} variant="top" src={icon} /> : <></>}
        </div>
      </div>
      <Card.Body>
        <Card.Title style={{cursor: "pointer"}}>{name}</Card.Title>
      </Card.Body>
    </Card>
  )
}

export default LocationCard;