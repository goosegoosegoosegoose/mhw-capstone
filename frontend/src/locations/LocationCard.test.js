import { render } from '@testing-library/react';
import LocationCard from "./LocationCard";
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <LocationCard />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});