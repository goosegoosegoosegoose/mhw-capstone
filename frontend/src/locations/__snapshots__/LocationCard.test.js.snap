// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3 mx-3 card bg-dark"
    style="width: 30vw; cursor: pointer;"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-sm-11 pt-2 mt-2"
      />
    </div>
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
        style="cursor: pointer;"
      />
    </div>
  </div>
</DocumentFragment>
`;
